exercises/affine-cipher/example.R

```diff
@@ -0,0 +1,63 @@
+# encrypts message
+encrypt <- function(message, a, b) {
+  m <- 26
+
+  # computes the greatest common divisor of numbers x and y
+  gcd <- function(x, y) {
+    r <- x %% y
+    return(ifelse(r, gcd(y, r), y))
+  }
+
+  # must check a and m are coprime
+  if (gcd(a, m) != 1) {
+    stop("a and 26 must be co-prime")
+  }
+
+  # removed whitespace & lower-cased
+  parsedmessage <- tolower(gsub(" ", "", message))
+  # list of letters
+  splitlist <- strsplit(parsedmessage, "")[[1]]
+  # index of letters
+  x <- match(splitlist, letters) - 1
+
+  # E(x) = (ax + b) mod m
+  return(paste(letters[ ((a * x + b) %% m) + 1], collapse = ""))
+}
+
+# decrypts encryption
+decrypt <- function(encryption, a, b) {
+  m <- 26
+
+  # computes the greatest common divisor of numbers x and y
+  gcd <- function(x, y) {
+    r <- x %% y
+    return(ifelse(r, gcd(y, r), y))
+  }
+
+  # must check a and m are coprime
+  if (gcd(a, m) != 1) {
+    stop("a and 26 must be co-prime")
+  }
+
+  # computes the modulo multiplicative inverse of a^m
+  # a^-1 = mmi( a mod m)=mmi(a,m)
+  mmi <- function(a, m) {
+    a <- a %% m
+    for (x in 1:m) {
+      if ((a * x) %% m == 1) {
+        return(x)
+      }
+    }
+    return(1)
+  }
+
+  # removed whitespace
+  parsedencryption <- gsub(" ", "", encryption)
+  # list of letters
+  splitlist <- strsplit(parsedencryption, "")[[1]]
+  # index of letters
+  y <- (match(splitlist, letters) - 1)
```

Comment:
This block (from L54) seems like a duplicate of L16-21 to me. Can we extract it as a helper function? It would to demonstrate the DRY principle to learners.

Reply:
Yes okay